packages/core/upload/admin/src/pages/App/index.js

```diff
@@ -1,31 +1,54 @@
-import React, { useEffect } from 'react';
-import { Helmet } from 'react-helmet';
+import React, { useEffect, Suspense } from 'react';
+import { Switch, Route } from 'react-router-dom';
 import { useIntl } from 'react-intl';
-import { useQueryParams } from '@strapi/helper-plugin';
-import { getTrad } from '../../utils';
+import { Helmet } from 'react-helmet';
+import { LoadingIndicatorPage, useQueryParams, useFocusWhenNavigate } from '@strapi/helper-plugin';
+import { Main } from '@strapi/design-system/Main';
+
 import { MediaLibrary } from './MediaLibrary';
+import { ConfigureTheView } from './ConfigureTheView';
+import { getTrad } from '../../utils';
+import pluginID from '../../pluginId';
+import { useConfig } from '../../hooks/useConfig';
+
+const Upload = () => {
+  const {
+    config: { isLoading, isError, data: config, error },
+  } = useConfig();
 
-const App = () => {
   const [{ rawQuery }, setQuery] = useQueryParams();
   const { formatMessage } = useIntl();
   const title = formatMessage({ id: getTrad('plugin.name'), defaultMessage: 'Media Library' });
 
   useEffect(() => {
-    if (!rawQuery) {
-      setQuery({ sort: 'updatedAt:DESC', page: 1, pageSize: 10 });
+    if (isLoading || isError || rawQuery) {
+      return;
     }
-  }, [rawQuery, setQuery]);
+    setQuery({ sort: config.data.sort, page: 1, pageSize: config.data.pageSize });
+  }, [isLoading, isError, config, error, rawQuery, setQuery]);
 
-  if (rawQuery) {
-    return (
-      <>
-        <Helmet title={title} />
-        <MediaLibrary />
-      </>
-    );
-  }
+  useFocusWhenNavigate();
 
-  return null;
+  return (
+    <Main aria-busy={isLoading}>
+      {isLoading && <LoadingIndicatorPage />}
+      {rawQuery ? (
+        <>
+          <Helmet title={title} />
+          <Suspense fallback={<LoadingIndicatorPage />}>
+            <Switch>
+              <Route exact path={`/plugins/${pluginID}`} component={MediaLibrary} />
+              <Route
+                exact
+                path={`/plugins/${pluginID}/configuration`}
+                render={() => <ConfigureTheView config={config} />}
```

Comment:
Would it be possible to lazy-load this component using `React.lazy`? If you search for `lazy` in the codebase you can find several usages. I think it would be quite beneficial here, because the page won't be used often and therefore a separate chunk can help not to increase the bundle size ...